modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/util/KBDropdownItemsProvider.java

```diff
@@ -802,6 +950,40 @@ private boolean _hasViewPermission(KBTemplate kbTemplate) throws Exception {
 		return false;
 	}
 
+	private boolean _isKBArticleSelected(KBArticle kbArticle) {
+		long resourceClassNameId = ParamUtil.getLong(
+			_liferayPortletRequest, "resourceClassNameId");
+
+		if (resourceClassNameId == kbArticle.getClassNameId()) {
+			long resourcePrimaryKey = ParamUtil.getLong(
+				_liferayPortletRequest, "resourcePrimKey",
+				KBArticleConstants.DEFAULT_PARENT_RESOURCE_PRIM_KEY);
+
+			if (resourcePrimaryKey == kbArticle.getResourcePrimKey()) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	private boolean _isKBFolderSelected(KBFolder kbFolder) {
+		long parentResourceClassNameId = ParamUtil.getLong(
+			_liferayPortletRequest, "parentResourceClassNameId");
+
+		if (parentResourceClassNameId == kbFolder.getClassNameId()) {
+			long parentResourcePrimaryKey = ParamUtil.getLong(
+				_liferayPortletRequest, "parentResourcePrimKey",
+				KBFolderConstants.DEFAULT_PARENT_FOLDER_ID);
+
+			if (parentResourcePrimaryKey == kbFolder.getKbFolderId()) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
```

Comment:
See my previous comment. If it is possible, try to flatten the conditional.

Reply:
I will also do it